src/services/semantic_cache.py

```
@@ -0,0 +1,88 @@
+                self._init_failed = True
+        return self._cache
+
+    async def get(self, prompt: str) -> str | None:
+        """Return a cached response for the prompt, or None if not found."""
+        cache = await self._get_cache()
+        if cache is None:
+            return None
+        try:
+            result = cache.lookup(prompt, llm_string="axioma-langflow")
+            if result:
+                prompt_hash = hashlib.md5(prompt.encode()).hexdigest()[:8]
```

Ruff (0.15.14)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)